jni/include/commons.h

```diff
@@ -45,8 +45,9 @@ namespace knn_jni {
          * address returned by {@link JNICommons#storeVectorData(long, float[][], long, long)}
          *
          * @param memoryAddress address to be freed.
+         * @param isBinaryIndex flag to indicate if the memory address is for binary index.
          */
-        void freeVectorData(jlong);
+        void freeVectorData(jlong, jboolean);
```

Comment:
It shouldn't have a concept of binary. It should be a type of underline memory. Like if it is float or uint8_t.

Reply:
Updated to call the right jni method in java layer.